syntax = "proto3";

package Ydb.PersQueue.ErrorCode;
option java_package = "com.yandex.ydb.persqueue.errorcode";

//500000..500999
enum ErrorCode {

    OK = 0;

    INITIALIZING = 500001;
    OVERLOAD = 500002;
    BAD_REQUEST = 500003;
    WRONG_COOKIE = 500004;
    SOURCEID_DELETED = 500024;

    WRITE_ERROR_PARTITION_IS_FULL = 500005;
    WRITE_ERROR_PARTITION_INACTIVE = 500029;
    WRITE_ERROR_DISK_IS_FULL = 500015;
    WRITE_ERROR_BAD_OFFSET = 500019;

    CREATE_SESSION_ALREADY_LOCKED = 500006;
    DELETE_SESSION_NO_SESSION = 500007;

    // removed READ_ERROR_IN_PROGRESS = 500008;
    READ_ERROR_NO_SESSION = 500009;
    // READ_TIMEOUT = 10;
    READ_ERROR_TOO_SMALL_OFFSET = 500011;
    READ_ERROR_TOO_BIG_OFFSET = 500012;

    SET_OFFSET_ERROR_COMMIT_TO_FUTURE = 500013;
    SET_OFFSET_ERROR_COMMIT_TO_PAST = 500025;

    TABLET_IS_DROPPED = 500014;

    READ_NOT_DONE = 500016;

    UNKNOWN_TOPIC = 500017;

    ACCESS_DENIED = 500018;
    CLUSTER_DISABLED = 500020;

    WRONG_PARTITION_NUMBER = 500021;

    // CREATE_TIMEOUT = 22;
    // IDLE_TIMEOUT = 23;

    PREFERRED_CLUSTER_MISMATCHED = 500026;

    TABLET_PIPE_DISCONNECTED = 500027;

    SCHEMA_ERROR = 500028;

    UNKNOWN_TXID = 500030;

    PRECONDITION_FAILED = 500031;

    ERROR = 500100;

    INVALID_ARGUMENT = 500040;
    VALIDATION_ERROR = 500080;
}
